cpp/CMakeLists.txt

```diff
@@ -217,6 +218,7 @@ target_link_libraries(
             CUDA::cusolver${_ctk_static_suffix}
             CUDA::cusparse${_ctk_static_suffix}
             $<$<BOOL:${RAFT_ENABLE_thrust_DEPENDENCY}>:raft::Thrust>
+            nvidia::cutlass::cutlass
```

Comment:
This dependency needs to be on the raft-distance target and not the raft target

Reply:
@robertmaynard I see that several tests and core algorithms depend on cosine/euclidean distance headers where we are using cutlass, hence I think it needs to be dependency on raft target. without it I am seeing several build failures when those sources are built.
I've modified get_cutlass.cmake from `raft-distance-exports` to `raft-exports`. Can this resolve the build issue in pylibraft? 
I've submitted this change and waiting to see if CI passes.

Reply:
@cjnolet Are you okay with cutlass being a hard requirement for raft?

Reply:
@mdoijade any tests/benchmarks and downstream projects which depend on distances also specify and use the `raft::distance` target.

Reply:
I see,  I've tried to add dependency on `raft_distance` wherever required and got the build working locally, please review if it looks good now.
also I've enabled the cutlass path only till cuda 11.x.